R/SSplotTags.R

```diff
@@ -43,380 +43,399 @@
+      # a function to plot the "tags alive" matrix
+      xvals <- as.numeric(substring(names(tagsalive)[-1], 7))
+      matplot(xvals, t(tagsalive[, -1]), type='l', lwd=3,
+              col=rich.colors.short(nrow(tagsalive)),
+              xlab="Period at liberty",
+              ylab="Estimated number of alive tagged fish",
+              main="'Tags alive' by tag group")
+      abline(h=0, col='grey')
+    }
+    
+    tagfun8 <- function(){
+      # a function to plot the "total recaptures" matrix
+      xvals <- as.numeric(substring(names(tagtotrecap)[-1], 7))
+      matplot(xvals, t(tagtotrecap[, -1]), type='l', lwd=3,
+              col=rich.colors.short(nrow(tagtotrecap)),
+              xlab="Period at liberty",
+              ylab="Estimated number of recaptures",
+              main="'Total recaptures' by tag group")
+      abline(h=0, col='grey')
+    }
+    
+    tagdata <- replist$tagdbase1
+    tagdata$Fleet <- as.character(tagdata$Fleet)
+    
+    max_num_fleets <- max(as.numeric(c(unique(tagdata$Fleet)))) #need to get the max number of fleets you have so you can rep over that. 
+    expected_by_fleets <- as.data.frame(rep(tagdbase2$Exp, each=max_num_fleets)) #make a new column to bind to your other frame that contains the breakdown of obs + exp recaps by fleet
+    names(expected_by_fleets)[1] <- "Expected" #rename column
+    new_tagdata <- cbind(tagdata, expected_by_fleets) #bind new column to tagdata dataframe
+    fleet_numbers <- new_tagdata %>% 
+      mutate(Numbers_Obs = round(Obs * N),
+             Numbers_Exp = round(Exp * Expected)) #generate exp. and obs. recaptures by fleet
+    fleet_numbers2 <- fleet_numbers %>%  
+      group_by(Fleet, Yr) %>% 
+      summarize(sum_exp = sum(Numbers_Exp), sum_obs = sum(Numbers_Obs))
+    
+    fleet_numbers2$Fleet <- sort(as.numeric(fleet_numbers2$Fleet), decreasing = FALSE)
+    fleet_numbers2$fleet_title <- paste("Fleet_", as.character(fleet_numbers2$Fleet),  sep="")
+    
+    mycols <- rep("black", max_num_fleets) #set colors for plotting expected values
+    myfill <- rep("gray35", max_num_fleets)
+    
+    tagfun9 <- function(){
+      #summarized observed and expected numbers of recaptures by fleet
+      fleet_plot2 <- ggplot(fleet_numbers2, aes(x=Yr, y=sum_obs)) +
+        geom_bar(aes(fill=as.factor(Fleet)), stat = "identity", alpha=0.5) +
+        geom_line(aes(y=sum_exp, color=as.factor(Fleet)), linetype=1, size=1) +
+        facet_wrap(fct_inorder(as.factor(fleet_numbers2$fleet_title)), scales="free") +
+        scale_fill_manual(values=myfill) +
+        scale_color_manual(values=mycols) +
+        labs(x="Year", y="Number of recaptures", subtitle="Observed (bar) and expected (line)") +
+        theme(strip.background = element_blank(), legend.position = "none")
+      print(fleet_plot2)
+    }
+    
+    #Calculate Pearson Residuals by fleet
+    fleetnumbers_PRs <- fleet_numbers %>% 
+      mutate(Pearson = (Numbers_Obs - Numbers_Exp) / sqrt(Numbers_Exp))
```

Comment:
I think (but could be wrong) that this is the only time dplyr is used. Since it is not currently a dependency of r4ss, I suggest changing this call to use just base r functionality and avoid making dplyr a dependency:
```r
fleetnumber_PRs <- fleet_numbers
fleetnumber_PRs[["Pearson"]] <- (fleetnumber_PRs[["Numbers_Obs"]] - 
  fleetnumber_PRs[["Numbers_Exp"]]) / sqrt(fleetnumber_PRs[["Numbers_Exp"]]))
```

It is definitely not as elegant (I am also a big fan of the dplyr syntax for data analysis!), but saves requiring users to download an additional package.

Reply:
Ian thought it was a good idea add the tidyverse packages, so I am working on adding them now!